chef/cookbooks/crowbar/templates/default/crowbar-shutdown-services-before-upgrade.sh.erb

```diff
@@ -11,6 +11,29 @@ echo "Executing $BASH_SOURCE"
 
 set -x
 
+<% if @cinder_controller %>
+
+set +x
+source /root/.openrc
+set -x
+
+# we need to stop scheduler so we can remove it from list of cinder services
+<% if @use_ha %>
+crm --wait resource stop cinder-scheduler
+<% else %>
+/etc/init.d/openstack-cinder-scheduler stop
+<% end %>
+
+# FIXME we need to do this only after all cinder-volume services are stopped!
+for service in cinder-volume cinder-scheduler; do
+    hosts=$(cinder service-list --binary $service| grep $service | sed 's/ //g' | cut -d "|" -f 3)
```

Comment:
Is there a special reason why you use ```cinder``` to list here and ```cinder-manage``` to remove? AFAICT you can do both steps with ```cinder-manage```. Which also has the advantage of not needing the cinder-api to run (it works by accessing the db models directly AFAIK). That would mean you could move the code to a later point, where the cluster (apart from the database) is already shutdown.